src/core/thread/mle_router.cpp

```diff
@@ -2035,6 +2035,7 @@ ThreadError MleRouter::HandleChildUpdateRequest(const Message &aMessage, const I
     }
 
     child->mLastHeard = Timer::GetNow();
+    child->mAddSrcMatchEntryShort = true;
```

Comment:
We are now discussing this issue within the Thread Group.  I'm fine with leaving this as is for now, but I think we should consider setting `child->mAddSrcMatchEntryShort` to `true` whenever we receive a message with the child's short address that passes MAC authentication.  I don't think a sleepy is required to send a Child Update Request message immediately after attaching to the network in all situations.